diff --git a/loader/main.c b/loader/main.c
index f8b113f..cd153e5 100644
--- a/loader/main.c
+++ b/loader/main.c
@@ -155,7 +155,9 @@ static void *try_dlopen( const char *argv0 )
     if (!argv0) return NULL;
     if (!(dir = realpath_dirname( argv0 ))) return NULL;

-    if ((p = remove_tail( dir, "/loader" )))
+    if ((p = remove_tail( dir, "/WineGame.app/Contents/MacOS" )))
+        path = build_path( p, "ntdll.so" );
+    else if ((p = remove_tail( dir, "/loader" )))
         path = build_path( p, "dlls/ntdll/ntdll.so" );
     else
         path = build_path( dir, "ntdll.so" );
